src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -277,7 +321,38 @@ protected async Task<(Guid, string projectName, SourceText)> GetGuidAndProjectNa
 
                 return new DocumentSpanEntry(
                     this, definitionBucket, documentSpan, spanKind,
-                    projectName, guid, sourceText, classifiedSpansAndHighlightSpan);
+                    projectName, guid, sourceText, classifiedSpansAndHighlightSpan, GetAggregatedCustomColumnsData());
+
+                // Local functions.
+                ImmutableDictionary<string, string> GetAggregatedCustomColumnsData()
```

Comment:
Local functions concern me a bit, esp. since we have no way to say "do not capture".  The primary concern here is that this function can be called *a lot*, so avoiding closure allocation is nice.  If possible, could you either confirm it doesn't capture, or can you extract into helper?

Reply:
Note: i will feel very different once "static local functions" are available.  Then, use them to your heart's content :)